### PR TITLE
CC compat

### PR DESCRIPTION
**Pull request by malte0811**
Don't worry about the 4,594 deletions, that's mostly the 1.12 compat code for OC and CT that I removed from `oldjava` since we have new implementations of both now.

As I said before I'm roughly following pup's idea of having one generic "can be called by a computer" concept which should be easy to use for other computer mods in the future. It does involve some weird `MethodHandle`-code, but that part can be treated as a black box most of the time.

### CODE REVIEW

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/cctweaked/ComputerCraftCompatModule.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/*
+ * BluSunrize
+ * Copyright (c) 2020
+ *
```
Comment (BluSunrize) [inline src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/cctweaked/ComputerCraftCompatModule.java:4]: Should these files not use a 2021 copyright statement?

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/generic/IndexArgument.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates that the annotated parameter is an index which is 1-based in Lua and 0-based in Java. The index will be
+ * automatically shifted when calling the method.
+ */
```
Comment (BluSunrize) [inline src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/generic/IndexArgument.java:19]: I continue hate Lua for this. But I'm glad you have a solution for it.

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/generic/impl/PoweredMBCallbacks.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+public class PoweredMBCallbacks extends Callback<PoweredMultiblockTileEntity<?, ?>>
+{
+	public static final PoweredMBCallbacks INSTANCE = new PoweredMBCallbacks();
+
+	public PoweredMBCallbacks()
+	{
+		addAdditional(EnergyCallbacks.INSTANCE);
+		addAdditional(MultiblockCallbacks.INSTANCE);
+	}
+	//TODO general access to recipes?
```
Comment (BluSunrize) [inline src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/generic/impl/PoweredMBCallbacks.java:23]: Hmm. Unsure on that. Seems very potent.

**Review by BluSunrize — APPROVED** [review]
LGTM.
No license statements on any of the new files. Is that intentional?

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/cctweaked/ComputerCraftCompatModule.java (hunk 1 above)
Comment (malte0811) [reply src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/cctweaked/ComputerCraftCompatModule.java:4]: This one isn't new, I just moved it and then modified it enough for git to call it new. I can change the copyright year on it if you want. The other files not having any copyright statement is definitely not intentional, I must have lost some configuration in IntelliJ at some point.

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/generic/impl/PoweredMBCallbacks.java (hunk 3 above)
Comment (malte0811) [reply src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/generic/impl/PoweredMBCallbacks.java:23]: I was talking about access to the current recipe, so you could e.g. make a display "currently converting plant oil and ethanol to biodiesel at a ratio of 2:2:4". That seems reasonable from a potency perspective to me. But I've essentially put that on the list of "implement this once I find an elegant approach for it".

Comment (BluSunrize) [reply src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/generic/impl/PoweredMBCallbacks.java:23]: Ah, I see

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/cctweaked/ComputerCraftCompatModule.java (hunk 1 above)
Comment (BluSunrize) [reply src/main/java/blusunrize/immersiveengineering/common/util/compat/computers/cctweaked/ComputerCraftCompatModule.java:4]: Right, it showed as new file, I didn't realize it was one of the moved ones.

**Pull request merged**